numpy/core/src/multiarray/compiled_base.c

```diff
@@ -1703,6 +1703,9 @@ pack_bits(PyObject *input, int axis, char order)
 static PyObject *
 unpack_bits(PyObject *input, int axis, PyObject *count_obj, char order)
 {
+    static int unpack_init = 0;
+    static npy_uint8 unpack_lookup_l[256][8];
```

Comment:
Valuable to tell GCC to align these as 8?

Reply:
gcc should do that by itself, at least on 64 bit platforms
also if it does not, we are using memcpy to access this on a 8 byte boundary (so we don't need to care about output alignment) so it won't break even if it is not.

one could enforce it portably be making this a int64 variable as before and casting back to char in the init loop, should we do that?

Reply:
I'm worried only about performance here, I agree it'll do the right thing with or without alignment.

I don't think the c standard requires GCC to align this, but perhaps it's clever enough to anyway. You could perhaps declare it an array of unions of u8[8] and u64, which would also allow you to construct one table from the other using bswap, and is a little cleaner that doing the casting you're suggesting.